dali/imgcodec/parsers/webp.cc

```diff
@@ -12,19 +13,157 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+#include <string>
 #include "dali/imgcodec/parsers/webp.h"
+#include "dali/core/byte_io.h"
 
 namespace dali {
 namespace imgcodec {
 
+namespace {
+
+// Chunk specification:
+// https://developers.google.com/speed/webp/docs/riff_container#riff_file_format
+//
+// Header layout:
+// RiffHeader | vp8_header_t | <WebpLossyHeader/WebpLosslessHeader>
+//
+// The WebpLossyHeader and WebpLosslessHeader are selected based on a identifier,
+// respectively "VP8 " and "VP8L" in vp8_header_t. "VP8X" in vp8_header_t signifies
+// WebP extended file format, which is not supported.
+
+// Struct must be packed for reading them to work correctly
+#pragma pack(push, 1)
+
+struct RiffHeader {
+  std::array<uint8_t, 4> riff_text;
+  uint32_t file_size;
+  std::array<uint8_t, 4> webp_text;
+};
+static_assert(sizeof(RiffHeader) == 12);
+
+using vp8_header_t = std::array<uint8_t, 4>;
+
+// Simple file format (lossy)
+// https://datatracker.ietf.org/doc/html/rfc6386#section-9.1
+struct WebpLossyHeader {
+  uint32_t chunk_size;
+  std::array<uint8_t, 3> frame_tag;
+  std::array<uint8_t, 3> sync_code;
+};
+static_assert(sizeof(WebpLossyHeader) == 10);
+
+// Simple file format (lossless)
+// https://developers.google.com/speed/webp/docs/webp_lossless_bitstream_specification#2_riff_header
+struct WebpLosslessHeader {
+  uint32_t chunk_size;
+  uint8_t signature_byte;
+};
+static_assert(sizeof(WebpLosslessHeader) == 5);
+
+#pragma pack(pop)  // end of packing scope
+
+template<size_t N>
+constexpr std::array<uint8_t, N - 1> tag(const char (&c)[N]) {
+  std::array<uint8_t, N - 1> a;
```

Comment:
```suggestion
  std::array<uint8_t, N - 1> a{};
```
Otherwise it's not constexpr according to gcc 7.3.